From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:34:21 -0800
Subject: [PATCH] wifi-scripts: add acs_exclude_6ghz_non_psc hostapd config option

Support excluding non-PSC channels from ACS on 6GHz radios.

Declare acs_exclude_6ghz_non_psc in the wifi-device schema and emit it
from the device config generator in usr/share/ucode/wifi/hostapd.uc when
enabled.

Fixes: OWRT-8793

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/schema/wireless.wifi-device.json
+++ b/src/usr/share/schema/wireless.wifi-device.json
@@ -8,6 +8,11 @@
 			"description": "Can be used to increase (or decrease) the likelihood of a specific channel to be selected by the ACS algorithm",
 			"type": "string"
 		},
+		"acs_exclude_6ghz_non_psc": {
+			"description": "Exclude non-PSC channels from ACS on 6GHz radios",
+			"type": "boolean",
+			"default": false
+		},
 		"acs_exclude_dfs": {
 			"description": "Exclude DFS channels from ACS",
 			"type": "boolean",
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -522,6 +522,9 @@ function generate(config) {
 
 	append_vars(config, [ 'acs_chan_bias', 'acs_exclude_dfs', 'acs_num_scans', 'enable_background_radar' ]);
 
+	if (config.acs_exclude_6ghz_non_psc)
+		append('acs_exclude_6ghz_non_psc', 1);
+
 	/* TX Power */
 	append_vars(config, [ 'min_tx_power' ]);
 
